### PR TITLE
AZ-56: Refactor command line arguments handling in ChainParams

### PR DESCRIPTION
**Pull request by h4nsu**
Small refactor of `ChainParams` using some cool features of `StructOpt`. Slightly changes the behavior of how `dev` chain is treated so that it behaves like described in comments:  now `dev` chain **always** uses predefined set of accounts and complains if told to use others.

### CODE REVIEW

File: bin/node/src/chain_spec.rs [hunk 1]
```diff
@@ -253,9 +234,10 @@ pub fn config(
     let faucet_account: AccountId =
         hex!["eaefd9d9b42915bda608154f17bb03e407cbf244318a0499912c2fb1cd879b74"].into();
 
-    let chain_type = match chain_id {
-        DEVNET_ID => ChainType::Development,
-        _ => ChainType::Live,
+    let chain_type = if chain_params.dev_chain {
```
Comment (fbielejec) [inline bin/node/src/chain_spec.rs:237]: this is just my preference, but in almost every case I find pattern matching easier to read than if-else statements (less lines of code if we want to use a metric), but this is by no means me insisting on it

Comment (h4nsu) [reply bin/node/src/chain_spec.rs:237]: In my noobish intuition match is like a heavy duty weapon, whereas if-else is just... good old if-else ;) Btw I'm kinda surprised there's no if-else one-liner in Rust...

Comment (fbielejec) [reply bin/node/src/chain_spec.rs:237]: I'm an animal, I would match even on a boolean value. But I have this killer arg to always show, different project, different languages, but left is a piece of a Solidity contract code soup, right is the same logic replicated in a Lisp using pattern matching on bool arrays :smile:  

DUZAylGXkAAg1T6

Maybe it's just me but even if someone doesn't fully get the syntax there's a stark difference in readability

Comment (h4nsu) [reply bin/node/src/chain_spec.rs:237]: Ok, I guess you converted me. How do I sign up for Church of Match? ;)

File: bin/node/src/chain_spec.rs [hunk 2]
```diff
@@ -34,6 +34,8 @@ pub const WELL_KNOWNS_ACCOUNTS: [&str; 7] = [
     "Eve",
 ];
```
[the remark this answers is not in the document]
Comment (h4nsu) [reply bin/node/src/chain_spec.rs:35]: Don't ask me, I didn't put them there. My goal was to refactor the usage of structopt for handling command line parameters.  Our node is based on substrate node and these accounts were already present in substrate node template. We might want to remove them, but I think this is outside the scope of this task.

File: run_nodes.sh [hunk 3]
```diff
@@ -57,12 +57,12 @@ validator_ids_string="${validator_ids_string//${IFS:0:1}/,}"
...
 
 echo "Bootstrapping chain for nodes 0..$((N_VALIDATORS - 1))"
-./target/release/aleph-node bootstrap-chain --millisecs-per-block 2000 --session-period 40 --base-path "$BASE_PATH" --chain-id dev --account-ids "$validator_ids_string" > "$BASE_PATH/chainspec.json"
+./target/release/aleph-node bootstrap-chain --millisecs-per-block 2000 --session-period 40 --base-path "$BASE_PATH" --account-ids "$validator_ids_string" > "$BASE_PATH/chainspec.json"
 
 for i in $(seq "$N_VALIDATORS" "$(( N_VALIDATORS + N_NON_VALIDATORS - 1 ))"); do
   echo "Bootstrapping node $i"
   account_id=${account_ids[$i]}
-  ./target/release/aleph-node bootstrap-node --base-path "$BASE_PATH" --chain-id dev --account-id "$account_id"
+  ./target/release/aleph-node bootstrap-node --base-path "$BASE_PATH" --account-id "$account_id"
```
[the remark this answers is not in the document]
Comment (h4nsu) [reply run_nodes.sh:65]: This is not about the env config. IIUC "Dev Chain" is one of the "modes" in which you can spawn your chain and it originates from substrate as a way to quickly start a "blockchain sandbox". It is something completely different then our devnet, even though naming might be confusing.

File: bin/node/src/chain_spec.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (KazyCC) [reply bin/node/src/chain_spec.rs:35]: Ok, fair point. Noted as point for retro meeting to take a closer look at this.

File: run_nodes.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (KazyCC) [reply run_nodes.sh:65]: Ok, thanks for explanation, apparently I'm the one that was confused. Make sense for me.

**Review by KazyCC — APPROVED** [review]
All my comments resolved, looks good to me :)

**Pull request merged**